packages/tempus/src/Tempus_TimeEventComposite.hpp

```diff
@@ -0,0 +1,207 @@
+// @HEADER
+// ****************************************************************************
+//                Tempus: Copyright (2017) Sandia Corporation
+//
+// Distributed under BSD 3-clause license (See accompanying file Copyright.txt)
+// ****************************************************************************
+// @HEADER
+
+#ifndef Tempus_TimeEventComposite_decl_hpp
+#define Tempus_TimeEventComposite_decl_hpp
+
+// Teuchos
+#include "Teuchos_Time.hpp"
+
+
+namespace Tempus {
+
+
+/** \brief This composite TimeEvent loops over added TimeEvents.
+ *
+ *  Individual TimeEvents are executed in the order in which they
+ *  were added.
+ */
+template<class Scalar>
+class TimeEventComposite : virtual public TimeEventBase<Scalar>
+{
+public:
+
+  /// Constructor
+  TimeEventComposite()
+  {
+    this->setName("TimeEventComposite");
+  }
+
+  /// Destructor
+  virtual ~TimeEventComposite() {}
+
+  /// \name Basic methods
+  //@{
+    /// Test if time is near a TimeEvent (within tolerance).
+    virtual bool isTime(Scalar time) const
+    {
+      bool is_time = false;
+      for(auto& e : timeEvents_) {
+        if (e->isTime(time)) {
+          is_time = true;
+          break;
+        }
+      }
```

Comment:
This is only checking if `any` and not `all`?

Reply:
Yeah, for the composite it should only need to match one of the TimeEvents to return true.  At some point it might be nice to know which TimeEvent matches, but for now we just need that it matches at least one TimeEvent.